src/parquet_bridge.rs

```diff
@@ -129,29 +131,26 @@ impl From<CompressionOptions> for CompressionCodec {
         match codec {
             CompressionOptions::Uncompressed => CompressionCodec::UNCOMPRESSED,
             CompressionOptions::Snappy => CompressionCodec::SNAPPY,
-            CompressionOptions::Gzip => CompressionCodec::GZIP,
+            CompressionOptions::Gzip(_) => CompressionCodec::GZIP,
             CompressionOptions::Lzo => CompressionCodec::LZO,
-            CompressionOptions::Brotli => CompressionCodec::BROTLI,
+            CompressionOptions::Brotli(_) => CompressionCodec::BROTLI,
             CompressionOptions::Lz4 => CompressionCodec::LZ4,
             CompressionOptions::Zstd(_) => CompressionCodec::ZSTD,
             CompressionOptions::Lz4Raw => CompressionCodec::LZ4_RAW,
         }
     }
 }
 
-/// Represents a valid zstd compression level.
-#[derive(Debug, Eq, PartialEq, Hash, Clone, Copy)]
-pub struct ZstdLevel(i32);
+/// Defines valid compression levels.
+pub trait CompressionLevel<T: std::fmt::Display + std::cmp::PartialOrd> {
+    const MINIMUM_LEVEL: T;
+    const MAXIMUM_LEVEL: T;
 
-impl ZstdLevel {
-    /// Attempts to create a zstd compression level from a given compression level.
-    ///
-    /// Compression levels must be valid (i.e. be acceptable for [`zstd::compression_level_range`])
-    #[cfg(feature = "zstd")]
-    pub fn try_new(level: i32) -> Result<Self, Error> {
-        let compression_range = zstd::compression_level_range();
```

Comment:
I removed this call so the trait could be used across the different compression types